### PR TITLE
Remove prints from the regular DALI execution flow

### PR DESCRIPTION
**Pull request by JanuszL**
- disables unnecessary prints for non-debug build
- converts some prints from cout to cerr and exceptions
  where appropriate

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

## Category:

**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
- disables unnecessary prints for non-debug build
- converts some prints from cout to cerr and exceptions
  where appropriate

### Affected modules and functionalities:
- dali/core/dynlink_cuda.cc
- dali/core/dynlink_cufile.cc
- dali/operators/reader/loader/filesystem.cc
- dali/operators/reader/nvdecoder/dynlink_nvcuvid.cc
- dali/pipeline/util/worker_thread.h
- dali/util/nvml_wrap.cc

### Key points relevant for the review:
- if no print has been missed
- if conversion to an exception or the usage of cerr is appropriate

### Tests
- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-2661

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

File: dali/operators/reader/loader/filesystem.cc [hunk 1]
```diff
@@ -202,7 +204,9 @@ vector<std::string> traverse_directories(const std::string &file_root, const std
   }
   // sort file names as well
   std::sort(file_list.begin(), file_list.end());
+#ifndef NDEBUG
```
Comment (jantonguirao) [inline dali/operators/reader/loader/filesystem.cc:207]: `LOG_LINE`?

File: dali/operators/reader/loader/filesystem.cc [hunk 2]
```diff
@@ -114,7 +114,9 @@ vector<std::pair<string, int>> traverse_directories(const std::string &file_root
   }
   // sort file names as well
   std::sort(file_label_pairs.begin(), file_label_pairs.end());
+#ifndef NDEBUG
   printf("read %lu files from %lu directories\n", file_label_pairs.size(), entry_name_list.size());
```
[the remark this answers is not in the document]
Comment (JanuszL) [reply dali/operators/reader/loader/filesystem.cc]: Done

File: dali/operators/reader/loader/filesystem.cc (hunk 1 above)
Comment (JanuszL) [reply dali/operators/reader/loader/filesystem.cc]: Done

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

**Review by mzient — APPROVED** [review]
This looks like a reasonable middle ground between removing this extra output and keeping the code that produces it :+1:

**Pull request merged**